shared/teams/team/member/index.new.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as Container from '../../../util/container'
+import * as Chat2Gen from '../../../actions/chat2-gen'
+import * as ProfileGen from '../../../actions/profile-gen'
+
+type Props = {
+  teamID: Types.TeamID
+  username: string
+}
+
+class TeamMember extends React.Component<Container.RouteProps<Props>> {
+  static navigationOptions = (ownProps: Container.RouteProps<Props>) => ({
+    header: Container.isMobile
+      ? () => (
+          <TeamMemberHeader
+            teamID={Container.getRouteProps(ownProps, 'teamID', Types.noTeamID)}
+            username={Container.getRouteProps(ownProps, 'username', '')}
+          />
+        )
+      : undefined,
+    headerExpandable: true,
+    headerTitle: () => (
+      <TeamMemberHeader
+        teamID={Container.getRouteProps(ownProps, 'teamID', Types.noTeamID)}
+        username={Container.getRouteProps(ownProps, 'username', '')}
+      />
+    ),
+  })
+  render() {
+    return null
+  }
+}
+
+// exported for stories
+export const TeamMemberHeader = (props: Props) => {
+  const {teamID, username} = props
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+
+  const teamMeta = Container.useSelector(s => Constants.getTeamMeta(s, teamID))
+  const teamDetails = Container.useSelector(s => Constants.getTeamDetails(s, teamID))
+  const yourUsername = Container.useSelector(s => s.config.username)
+
+  const onBack = () => dispatch(nav.safeNavigateUpPayload())
+  const onChat = () =>
+    dispatch(Chat2Gen.createPreviewConversation({participants: [username], reason: 'memberView'}))
+  const onViewProfile = () => dispatch(ProfileGen.createShowUserProfile({username}))
+  const onViewTeam = () =>
+    dispatch(nav.safeNavigateAppendPayload({path: [{props: {teamID}, selected: 'team'}]}))
+
+  const member = teamDetails.members.get(username)
+  if (!member) {
+    // loading? should never happen.
```

Comment:
maybe log a warning here since who knows what bugs we'll encounter while we're ironing out this new stuff.